src/js/messaging/reducers/search.js

```diff
@@ -1,30 +1,47 @@
 import set from 'lodash/fp/set';
 
+import { makeField } from '../../common/model/fields';
+
 import {
+  SEARCH_SUCCESS,
   SET_ADVSEARCH_END_DATE,
   SET_ADVSEARCH_START_DATE,
+  SET_SEARCH_PARAM,
   TOGGLE_ADVANCED_SEARCH
 } from '../utils/constants';
 
 const initialState = {
-  basic: null,
-  advanced: {
-    visible: false,
-    params: {
-      dateRange: {
-        start: null,
-        end: null
-      }
+  params: {
+    dateRange: {
+      start: null,
+      end: null
+    },
+    term: makeField(''),
+    from: {
+      field: makeField(''),
+      exact: false
+    },
+    subject: {
+      field: makeField(''),
+      exact: false
     }
+  },
+  advanced: {
+    visible: false
   }
 };
 
 export default function modals(state = initialState, action) {
   switch (action.type) {
-    case SET_ADVSEARCH_START_DATE:
-      return set('advanced.params.dateRange.start', action.date, state);
     case SET_ADVSEARCH_END_DATE:
-      return set('advanced.params.dateRange.end', action.date, state);
+      return set('params.dateRange.end', action.date, state);
+    case SET_ADVSEARCH_START_DATE:
+      return set('params.dateRange.start', action.date, state);
+    case SET_SEARCH_PARAM:
+      return set(action.path, action.field, state);
```

Comment:
Since it's always only affecting `params`, maybe set it on `params.${action.path}` and cut the `params.` prefix from the path arguments when calling the action creator.